google/terraform/compute_instance.go

```diff
@@ -0,0 +1,70 @@
+package terraform
+
+import (
+	"github.com/cycloidio/terracost/price"
+	"github.com/cycloidio/terracost/product"
+	"github.com/cycloidio/terracost/query"
+	"github.com/cycloidio/terracost/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/shopspring/decimal"
+)
+
+// ComputeInstance is the entity that holds the logic to calculate price
+// of the google_compute_instance
+type ComputeInstance struct {
+	provider     *Provider
+	region       string
+	instanceType string
+
+	machineType string
+}
+
+type computeInstanceValues struct {
+	MachineType string `mapstructure:"machine_type"`
+}
+
+// decodeComputeInstanceValues decodes and returns computeInstanceValues from a Terraform values map.
+func decodeComputeInstanceValues(tfVals map[string]interface{}) (computeInstanceValues, error) {
```

Comment:
commenting this one but not the others?

Reply:
I copied this code and just modified the needs from the aws one, I'll add the comments as it would be better.